VideoLocker/src/main/java/org/edx/mobile/receivers/NetworkConnectivityReceiver.java

```diff
@@ -33,7 +33,8 @@ public void onReceive(Context context, Intent intent) {
         NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
         boolean isConnected = activeNetwork != null && activeNetwork.isAvailable();
 
-        if(isConnected) {
+        // do NOT test download speed on Zero-Rated network
+        if(isConnected && !NetworkUtil.isOnZeroRatedNetwork(context)) {
             logger.debug("Have reconnected, testing download speed.");
             //start an instance of the download speed service so it can run in the background
             Intent speedTestIntent = new Intent(context, DownloadSpeedService.class);
```

Comment:
if the goal is not to check traffic on zero rated network,  NetworkUtil.isOnZeroRatedNetwork() check should be at the start of the method call